salt/utils/decorators/__init__.py

```diff
@@ -140,6 +142,29 @@ class depends(Depends):  # pylint: disable=C0103
     '''
 
 
+def timing(function):
+    '''
+    Decorator wrapper to log execution time, for profiling purposes
+    '''
+    def wrapped(*args, **kwargs):
```

Comment:
You should add a `functions.wraps(function)` decorator to `wrapped` to keep the `wrapped` function attributes equal to those of `function`...